src/webgpu/shader/validation/functions/restrictions.spec.ts

```diff
@@ -585,3 +585,43 @@ fn foo() {
 
     t.expectCompileResult(false, code);
   });
+
+g.test('param_names_must_differ')
+  .specURL('https://gpuweb.github.io/gpuweb/wgsl/#function-declaration-sec')
+  .desc(`Test that function parameters must have different names`)
+  .params(u => u.combine('p1', ['a', 'b', 'c'] as const).combine('p2', ['a', 'b', 'c'] as const))
+  .fn(t => {
+    const code = `fn foo(${t.params.p1} : u32, ${t.params.p2} : f32) { }`;
+    t.expectCompileResult(t.params.p1 !== t.params.p2, code);
+  });
+
+g.test('param_scope_is_function_body')
+  .specURL('https://gpuweb.github.io/gpuweb/wgsl/#function-declaration-sec')
+  .desc(`Test that function parameters are only in scope in the function body`)
+  .params(u => u.combine('use', ['body', 'var', 'const', 'override', 'function'] as const))
```

Comment:
nit: Unrolling this into an `Record<string, string>` with 5 source blocks would be a lot easier to read.

Reply:
Makes sense, will change.